DNN Platform/Library/WebControls/WebControlBase.cs

```diff
@@ -1,95 +1,123 @@
 
-    using DotNetNuke.Common;
+    using DotNetNuke.Common;
     using DotNetNuke.Entities.Portals;
 
-    public abstract class WebControlBase : WebControl
-    {
-        private string _styleSheetUrl = string.Empty;
-        private string _theme = string.Empty;
-
-        public string ResourcesFolderUrl
-        {
-            get
-            {
-                return Globals.ResolveUrl("~/Resources/");
-            }
-        }
-
-        public bool IsHostMenu
-        {
-            get
-            {
-                return Globals.IsHostTab(this.PortalSettings.ActiveTab.TabID);
-            }
-        }
-
-        [Browsable(false)]
-        [DesignerSerializationVisibility(DesignerSerializationVisibility.Hidden)]
-        public PortalSettings PortalSettings
-        {
-            get
-            {
-                return PortalController.Instance.GetCurrentPortalSettings();
-            }
-        }
-
-        public abstract string HtmlOutput { get; }
-
-        [Obsolete("There is no longer the concept of an Admin Page.  All pages are controlled by Permissions. Scheduled removal in v11.0.0.", true)]
-        public bool IsAdminMenu
-        {
-            get
-            {
-                bool _IsAdmin = false;
-                return _IsAdmin;
-            }
-        }
-
-        public string Theme
-        {
-            get
-            {
-                return this._theme;
-            }
-
-            set
-            {
-                this._theme = value;
-            }
-        }
-
-        public string StyleSheetUrl
-        {
-            get
-            {
-                if (this._styleSheetUrl.StartsWith("~"))
-                {
-                    return Globals.ResolveUrl(this._styleSheetUrl);
-                }
-                else
-                {
-                    return this._styleSheetUrl;
-                }
-            }
-
-            set
-            {
-                this._styleSheetUrl = value;
-            }
-        }
-
-        protected override void RenderContents(HtmlTextWriter output)
-        {
-            output.Write(this.HtmlOutput);
-        }
-    }
-}
+    /// <summary>
+    /// Base class for Dnn specific WebControls.
+    /// </summary>
+    public abstract class WebControlBase : WebControl
+    {
+        private string styleSheetUrl = string.Empty;
+        private string theme = string.Empty;
+
+        /// <summary>
+        /// Gets a url to the web control resources folder.
+        /// </summary>
+        public string ResourcesFolderUrl
+        {
+            get
+            {
+                return Globals.ResolveUrl("~/Resources/");
+            }
+        }
+
+        /// <summary>
+        /// Gets a value indicating whether the control is currently in the host menu.
+        /// </summary>
+        [Obsolete("There is no longer the concept of aa Host Page.  All pages are controlled by Permissions. Scheduled removal in v11.0.0.")]

```

Comment:
I'm not sure that it's true that there isn't a concept of a host page anymore.  It's still possible that have a page with a -1 portal ID, which would only be accessible to super users

Reply:
Makes sense, I removed the [Obsolete]